addons/web_hierarchy/static/src/hierarchy_model.js

```diff
@@ -939,24 +957,36 @@ export class HierarchyModel extends Model {
                     fetchParentChildren = true;
                     domain = Domain.or([domain, [[this.parentFieldName, "=", parentNode.resId]]]);
                 }
-                if (node.id === node.tree.root.id) {
-                    this.root.removeTree(node.tree);
-                    node.tree = parentNode.tree;
-                } else {
-                    node.removeParentNode();
-                }
-            } else {
-                node.removeParentNode();
-                this.root.addNewRootNode(node);
             }
+            node.hidden = true;
             this.notify({ scrollTarget: "none" });
             await this.mutex.exec(async () => {
-                await this.orm.write(
-                    this.resModel,
-                    [node.resId],
-                    { [this.parentFieldName]: parentResId || parentNode?.resId || false },
-                    { context: this.config.context }
-                );
+                try {
+                    await this.updateParentId(node, parentResId || parentNode?.resId);
+                    node.hidden = false;
+                    if (parentNode) {
+                        if (node.id === node.tree.root.id) {
+                            this.root.removeTree(node.tree);
+                            node.tree = parentNode.tree;
+                        } else {
+                            node.removeParentNode();
+                        }
+                    } else {
+                        node.removeParentNode();
+                        this.root.addNewRootNode(node);
+                    }
+                } catch (error) {
+                    node.hidden = false;
+                    this.notify({ scrollTarget: "none" });
+                    if (error.exceptionName === "odoo.exceptions.ValidationError") {
+                        this.notification.add(
+                            error.data.message,
+                            { type: "danger" },
+                        )
+                        return;
+                    }
```

Comment:
I am wondering if that part should not be a action returned by a custom method in knowledge to display a danger notification instead of generically taking into account all `odoo.exceptions.ValidationError` as a notification. 🤔 

That is, in knowledge or whatever, we could alter the `updateParentId` to change the rpc call and then using the res to display or not a notification. 🤷🏼‍♂️ 

What do you think? 🙂 

In the case, you don't know we can return an action to display a notification: 
https://github.com/odoo/odoo/blob/1552bdd5fb5ff38034b0cecba96e982e9c935240/addons/sale_project/models/sale_order.py#L175-L182

and when the JS code will receive that object we can use `doAction` in action service to execute the action to display the notification. 🙂

Reply:
I removed the notification part, I'll let the rpc in knowledge throw an error and it'll be displayed in a popup instead of a notification, like other errors, that way I don't have to add error handling in the hierarchy model